### ISSUE TITLE
[Bug Report][3.5.2] v-stepper "non-linear" property not working as expected 

### ISSUE THREAD

**Issue by kunalmah** [vuetifyjs/vuetify#19275]
### Environment
**Vuetify Version:** 3.5.2
**Vue Version:** 3.4.15
**Browsers:** Chrome 121.0.0.0
**OS:** Windows 10

### Steps to reproduce
With "non-linear" property, ideally, user should be allowed to move through the steps when they click on the v-stepper-header. However, the property seems to have no effect. Only when "editable" is added, user is allowed to move non-linearly by clicking on the header.

### Expected Behavior
Adding "non-linear" alone to the v-stepper should allow user to move from one step to another by clicking on the v-stepper-header.

### Actual Behavior
Only when "editable" property is added, user can click on the steps to move non-linearly. "non-linear" seems to have no effect.

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNrtVMFKxDAQ/ZUxlyrYlt11L9IteBA8qBcPHqyH2M5qIG1CkhZl2X83SbclhYIK4kH2NvPmzXtpZ5inHdGqTK+kTLoWySXJDNaSU4N50QBkXUyl9KFPStEYyhpUB6gHqarGfERiwwzHHO5FA7e2hSroYm1QSlSQpRPSTDO+mwD2haG5EU3MveKEEXLiN6RV8MoZBrMfCt5+U5C7D3iwOCwKAh3lrcNsnLt3hg1zghXrmPNy3CH+qe8y8LXxn/muAl8bf+Ublmf+cFAORzrOOpzpgH5/ka4rZugLx99aIzzoHZfovyyRTycXyiH9AcvS4LDZVJeKSQMaTdvfN1ZLoQzsQOEW9rBVoobI3sTI0QGsrjZQ61fYOMZpdIOcC3gUilcn0Zkz6CWtGNmfk1VykSzWxAXrZEmePwGZZp1v)

**Comment by OleKal** [comment]
Why was this bug closed? This is still an issue

**Comment by J-Sek** [comment]
Yup... docs could be improved. All this prop does is +50% bump in opacity to show unfinished items as active. The section about "Linear steppers" could also be more precise as there is no internal logic enforcing linear/non-linear
